packages/lodestar/src/network/util.ts

```diff
@@ -61,23 +61,27 @@ export function isRequestSingleChunk(method: Method): boolean {
   return Methods[method].responseType === MethodResponseType.SingleResponse;
 }
 
-export function eth2ResponseTimer<T>(): (source: AsyncIterable<T>) => AsyncGenerator<T> {
-  return (source) => {
-    return (async function*() {
-      const controller = new AbortController();
-      let responseTimer = setTimeout(() => controller.abort(), TTFB_TIMEOUT);
-      const renewTimer = (): void => {
-        clearTimeout(responseTimer);
-        responseTimer = setTimeout(() => controller.abort(), RESP_TIMEOUT);
-      };
-      const cancelTimer = (): void => {
-        clearTimeout(responseTimer);
-      };
-      for await(const item of abortSource(source, controller.signal, {abortMessage: "response timeout"})) {
-        renewTimer();
-        yield item;
-      }
-      cancelTimer();
-    })();
+export function initResponseTimer<T>(): (() => ((source: AsyncIterable<T>) => AsyncGenerator<T>)) {
+  const controller = new AbortController();
+  let responseTimer = setTimeout(() => controller.abort(), TTFB_TIMEOUT);
+  const renewTimer = (): void => {
+    clearTimeout(responseTimer);
+    responseTimer = setTimeout(() => controller.abort(), RESP_TIMEOUT);
   };
-}
+  const cancelTimer = (): void => {
+    clearTimeout(responseTimer);
+  };
+  const checkTimeout = (): ((source: AsyncIterable<T>) => AsyncGenerator<T>) => {
+    return (source) => {
+      return (async function*() {
+        const abortOpt = {abortMessage: "response timeout"};
+        for await(const item of abortSource(source, controller.signal, abortOpt)) {
+          renewTimer();
```

Comment:
It seems that the RESP_TIMEOUT is 'The maximum time for complete response transfer.'
Shouldn't be reset, but rather tracked across the whole response

Reply:
I think RESP_TIMEOUT is for per response_chunk as in the p2p spec: `The requester MUST wait a maximum of TTFB_TIMEOUT for the first response byte to arrive (time to first byte—or TTFB—timeout). On that happening, the requester allows a further RESP_TIMEOUT for each subsequent response_chunk received.`

Reply:
I also see `The maximum time for complete response transfer.`, this is misleading

Reply:
Gotcha, you wanna make a PR to the spec to clarify?

Reply:
probably not at this time, it's too small and all clients understood it I think :)